Src/EB/AMReX_EB2_3D_C.H

```diff
@@ -4,6 +4,40 @@
 
 namespace amrex { namespace EB2 {
 
+namespace {
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void cover_cut_edges_given_corner (
+                       int i, int j, int k,
+                       Array4<Type_t> const& ex,
+                       Array4<Type_t> const& ey,
+                       Array4<Type_t> const& ez)
+   {
+      //x edges
+      if (j >= ex.begin.y && j < ex.end.y && k >= ex.begin.z && k < ex.end.z) {
+         if (i > ex.begin.x && i <= ex.end.x && ex(i-1,j,k) == Type::irregular)
+            ex(i-1,j,k) = Type::covered;
+         if (i >= ex.begin.x && i < ex.end.x && ex(i  ,j,k) == Type::irregular)
+            ex(i,j,k) = Type::covered;
+      }
+
+      //y edges
+      if (i >= ey.begin.x && i < ey.end.x && k >= ey.begin.z && k < ey.end.z) {
+         if (j > ey.begin.y && j <= ey.end.y && ey(i,j-1,k) == Type::irregular)
+            ey(i,j-1,k) = Type::covered;
+         if (j >= ey.begin.y && j < ey.end.y && ey(i,j  ,k) == Type::irregular)
+            ey(i,j,k) = Type::covered;
+      }
+
+      //z edges
+      if (i >= ez.begin.x && i < ez.end.x && j >= ez.begin.y && j < ez.end.y) {
+         if (k > ez.begin.z && k <= ez.end.z && ez(i,j,k-1) == Type::irregular)
```

Comment:
I think you have to launch a separate kernel to do this.  The problem is `ez(i,j,k-1)` may not even have been initialized by another thread yet.